pages/libraryFiles/LibraryFileHandler.inc.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * @file pages/libraryFiles/LibraryFileHandler.inc.php
+ *
+ * Copyright (c) 2014-2018 Simon Fraser University
+ * Copyright (c) 2000-2018 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class LibraryFileHandler
+ * @ingroup pages_libraryFiles
+ *
+ * @brief Class defining a handler for library file access
+ */
+
+// Import the base handler.
+import('classes.handler.Handler');
+
+class LibraryFileHandler extends Handler {
+
+	//
+	// Public handler methods
+	//
+
+	/**
+	 * Download a library public file.
+	 * @param $args array
+	 * @param $request Request
+	 */
+	function downloadPublic($args, $request) {
+		import('classes.file.LibraryFileManager');
+		$context = $request->getContext();
+		$libraryFileManager = new LibraryFileManager($context->getId());
+		$libraryFileDao = DAORegistry::getDAO('LibraryFileDAO');
+
+		$publicFileId = $args[0];
+
+		$libraryFile = $libraryFileDao->getById($publicFileId);
+		if ($libraryFile && $libraryFile->getPublicAccess() && $libraryFile->getContextId() == $context->getId()) {
+			$filePath = $libraryFileManager->getBasePath() .  $libraryFile->getOriginalFileName();
+			$libraryFileManager->downloadFile($filePath);
+		} else {
+				header('HTTP/1.0 403 Forbidden');
+				echo '403 Forbidden<br>';
+				return;
+		}
+	}
+
+	/**
+	 * Download a library file.
+	 * @param $args array
+	 * @param $request Request
+	 */
+	function downloadLibraryFile($args, $request) {
+		import('classes.file.LibraryFileManager');
+		$context = $request->getContext();
+		$libraryFileManager = new LibraryFileManager($context->getId());
+		$libraryFileDao = DAORegistry::getDAO('LibraryFileDAO');
+		$libraryFile = $libraryFileDao->getById($request->getUserVar('libraryFileId'));
```

Comment:
This fetch is also missing a check for the context ID.

Generally we have a pattern to follow for this: getter functions like `getById` should have an optional second parameter for the contextId. When specified, it adds an `AND context_id = ?` to the SQL. That way supplying the second parameter becomes an automatic check for the right context ID. I'd suggest adding that option to the DAO code then using it wherever we're calling `LibraryFileDAO::getById`.

Reply:
@asmecher I made the change. I found that there was a documented parameter inconsistency, and I changed that too. (supposing that the other documented parameters don't serve any purpose for the current code version at least)

By making a quick search I think there are more such inconsistencies between function parameters and documented parameters, but I am not sure if that is on purpose.

For example 
https://github.com/pkp/pkp-lib/blob/master/classes/context/PKPSectionDAO.inc.php#L32
or 
https://github.com/pkp/pkp-lib/blob/master/classes/controlledVocab/ControlledVocabEntryDAO.inc.php#L35

Reply:
@defstat, thanks -- these self-documentation inconsistencies aren't intentional; feel free to quickly correct them when you see them.